comp/core/tagger/impl-dual/dual.go

```diff
@@ -0,0 +1,81 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package dualimpl contains the implementation of the dual tagger.
+// The dualimpl allow clients to use either the remote tagger or the local based on
+// their configuration
+package dualimpl
+
+import (
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	log "github.com/DataDog/datadog-agent/comp/core/log/def"
+	tagger "github.com/DataDog/datadog-agent/comp/core/tagger/def"
+	local "github.com/DataDog/datadog-agent/comp/core/tagger/impl"
+	remote "github.com/DataDog/datadog-agent/comp/core/tagger/impl-remote"
+	"github.com/DataDog/datadog-agent/comp/core/telemetry"
+	workloadmeta "github.com/DataDog/datadog-agent/comp/core/workloadmeta/def"
+	compdef "github.com/DataDog/datadog-agent/comp/def"
+	"github.com/DataDog/datadog-agent/pkg/util/optional"
+)
+
+// Requires contains the dependencies for the dual tagger component
+type Requires struct {
+	Lc           compdef.Lifecycle
+	LocalParams  tagger.Params
+	RemoteParams tagger.RemoteParams
+	DualParams   tagger.DualParams
+	Config       config.Component
+	Log          log.Component
+	Wmeta        optional.Option[workloadmeta.Component]
+	Telemetry    telemetry.Component
+}
+
+// Provides contains returned values for the  dual tagger component
+type Provides struct {
+	local.Provides
```

Comment:
Why don't we have a `remote` field here ?

Reply:
Because local provides cover the case for remote which only returns the component and he case of the local tagger we return the component and the API endpoint provider https://github.com/DataDog/datadog-agent/blob/21cb4d6aa90d6674f7b67c6209d09e41a56d9ac5/comp/core/tagger/impl/tagger.go#L78

So for the case of remote we won't provide an API endpoint provider